test/tracks.e2e-spec.ts

```diff
@@ -1,21 +1,61 @@
-import { INestApplication, ValidationPipe } from '@nestjs/common';
+import { Track } from './../src/tracks/models/track.entity';
+import { JwtPayload } from './../src/auth/models/jwt-payload';
+import { RolesGuard } from '../src/auth/guards/roles.guard';
+import {
+  ExecutionContext,
+  INestApplication,
+  ValidationPipe,
+  UnauthorizedException,
+  BadGatewayException,
+  BadRequestException,
+  ForbiddenException
+} from '@nestjs/common';
+import { ContextIdFactory } from '@nestjs/core';
 import { Test, TestingModule } from '@nestjs/testing';
+import {
+  getModelForClass,
+  mongoose,
+  ReturnModelType
+} from '@typegoose/typegoose';
 import * as request from 'supertest';
 
+import { JwtAuthGuard } from '../src/auth/guards';
+import { JwtStrategy } from '../src/auth/strategies/jwt.strategy';
 import { CreateTrackDto } from '../src/tracks/models/dto/create-track.dto';
 import { TracksModule } from '../src/tracks/tracks.module';
-import { DbTest } from './db-test.module';
-
-// process.env.API_AUTH_ENABLED = 'true';
+import { TracksService } from '../src/tracks/tracks.service';
+import { User, UserRole } from '../src/users/models/user.entity';
+import { DbTest, inMemoryDb } from './db-test.module';
 
 describe('TracksController (e2e)', () => {
   let app: INestApplication;
   let route: request.SuperTest<request.Test>;
-
+  let service: TracksService;
+  let mongo: typeof mongoose;
+  const validEmail = 'email@gmail.com';
+  const validPass = 'pass@45Pdd';
+  let currentUser: JwtPayload;
+  let TrackModel: ReturnModelType<typeof Track>;
+  const jwtGuard = {
+    canActivate: (context: ExecutionContext): boolean => {
+      throw new UnauthorizedException();
+    }
+  };
+  const rolesGuard = {
```

Comment:
Codacy found an issue: ['rolesGuard' is assigned a value but never used.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-backend/pullRequest?prid=5747194)